src/main/resources/application.properties

```diff
@@ -1,15 +1,21 @@
 server.port=8080
+
 workspace.maxStairwayThreads=4
 workspace.resourceId=mc-terra-workspace-manager
 workspace.stairwayTimeoutSeconds=1800
+
 db.workspace.uri=jdbc:postgresql://127.0.0.1:5432/${DATABASE_NAME}
 db.workspace.username=${DATABASE_USER}
 db.workspace.password=${DATABASE_USER_PASSWORD}
 db.workspace.initializeOnStart=false
 db.workspace.upgradeOnStart=true
+
 db.stairway.uri=jdbc:postgresql://127.0.0.1:5432/${STAIRWAY_DATABASE_NAME}
 db.stairway.username=${STAIRWAY_DATABASE_USER}
 db.stairway.password=${STAIRWAY_DATABASE_USER_PASSWORD}
 db.stairway.migrateUpgrade=false
 db.stairway.forceClean=true
+
 sam.basePath=${SAM_ADDRESS}
+
+datarepo.instances=https://jade.datarepo-dev.broadinstitute.org,https://jade-terra.datarepo-prod.broadinstitute.org,https://data.terra.bio
```

Comment:
I really think these should be key-value pairs, instead of raw values:
* terra-prod = "https://jade-terra.datarepo-prod.broadinstitute.org"
* terra-dev = "https://jade.datarepo-dev.broadinstitute.org"

When we persist data references to the db, we should persist them by key; e.g. the data reference would contain "terra-prod" as its instance value.

When Terra's prod data repo moves from "https://jade-terra.datarepo-prod.broadinstitute.org" to "https://data.terra.bio", we can simply update the value inside this config, and pre-existing references will use the new hostname. Else, we'd have to run a migration against all pre-existing references - or leave them broken - or ask data repo to maintain the old hostname forever.

As an additional benefit, the user-facing message inside the validation exception "Valid instances are ... " need only output the valid keys, so we keep our inner workings of hostnames abstracted from end users.

I realize that to do this easily in config, we might want to switch to [YAML instead of properties](https://docs.spring.io/spring-boot/docs/1.2.0.M1/reference/html/boot-features-external-config.html#boot-features-external-config-yaml) syntax.

I mentioned this in person (in passing) - feel free to push back and say it should be a follow-on PR.

Reply:
I suspect it's worth it, avoiding a migration is reason enough for me, but I agree with your other points, too. Thanks for the detailed explanation of your thoughts, it's really helpful!

Reply:
For the moment, I encoded the current prod and the future prod under different keys - this is because in my experience from the UI, there's an overlap period when the new DNS name is pointed and some people/configs still have the old url, so we can have both active at once and remove the current one at our leisure, or never. Your thoughts, @davidangb?

Reply:
Ah, just caught the second part of this that I didn't parse the first time, that we store the keys in our db. Let me take a pass at that.